Plutarch/Rec.hs

```diff
@@ -0,0 +1,104 @@
+module Plutarch.Rec (
+  PRecord (PRecord, getRecord),
+  ScottArgument,
+  ScottEncoded,
+  ScottEncoding,
+  accessors,
+  letrec,
+  (#.),
+) where
+
+import Control.Monad.Trans.State.Lazy (State, evalState, get, put)
+import Data.Functor.Compose (Compose)
+import Data.Kind (Type)
+import Data.Monoid (Dual (Dual, getDual), Endo (Endo, appEndo), Sum (Sum, getSum))
+import Numeric.Natural (Natural)
+import Plutarch (PCon, pcon, phoistAcyclic, plam, punsafeCoerce, (#), (:-->))
+import Plutarch.Internal (
+  RawTerm (RApply, RLamAbs, RVar),
+  Term (Term, asRawTerm),
+  TermResult (TermResult, getDeps, getTerm),
+  mapTerm,
+ )
+import qualified Rank2
+
+{- |
+  Highest precedence infixl operator, to be used like record field accessor. e.g.:
+
+  >>> record #. field
+-}
+(#.) ::
```

Comment:
We probably want to rename/remove this since it's not useful (yet?), and will likely have to have its type changed anyway if it does become useful.

Reply:
If I remove it only to inline its uses in tests, I'll end up with unsightly mess:

```
-        , testCase "even 5" $ equal' (evenOdd #. even # (5 :: Term s PInteger)) "(program 1.0.0 False)"
+        , testCase "even 5" $ equal' (punsafeCoerce evenOdd # getScott (even accessors) # (5 :: Term s PInteger)) "(program 1.0.0 False)"
```

Reply:
I've got something better now:

    testCase "even 5" $ equal' (evenOdd # field even # (5 :: Term s PInteger)) "(program 1.0.0 False)"